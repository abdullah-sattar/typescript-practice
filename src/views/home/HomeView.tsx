import Button from "../../components/button/Button";
import Layout from "../../components/layout/Layout";

export const HomeView = () => {
  return (
    <Layout>
      <h1>Page Heading</h1>
      <h2>Section Heading</h2>
      <h3>Panel Heading</h3>
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Non error
        dolor in nostrum minima odio a fuga saepe soluta adipisci perspiciatis
        maxime tempora, temporibus beatae voluptas repellat rerum. Dignissimos,
        necessitatibus.
      </p>
      <Button
        label="Log to console"
        onClick={() => {
          console.log("button clicked");
        }}
      />
    </Layout>
  );
};

export default HomeView;
