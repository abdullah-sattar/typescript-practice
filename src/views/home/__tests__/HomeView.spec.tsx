import { customRender } from "../../../utils/testUtils";
import HomeView from "../HomeView";

export default describe("HomeView", () => {
  it("Renders", () => {
    const { container } = customRender(<HomeView />);

    expect(container).toMatchSnapshot();
  });
});
