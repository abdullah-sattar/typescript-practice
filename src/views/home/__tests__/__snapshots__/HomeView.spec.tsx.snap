// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView Renders 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <h1>
        Page Heading
      </h1>
      <h2>
        Section Heading
      </h2>
      <h3>
        Panel Heading
      </h3>
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Non error dolor in nostrum minima odio a fuga saepe soluta adipisci perspiciatis maxime tempora, temporibus beatae voluptas repellat rerum. Dignissimos, necessitatibus.
      </p>
      <button
        class="button"
        type="button"
      >
        Log to console
      </button>
    </main>
  </div>
</div>
`;
